client/landing/stepper/declarative-flow/internals/steps-repository/pattern-assembler/index.tsx

```diff
@@ -64,7 +79,11 @@ const PatternAssembler: Step = ( { navigation, flow } ) => {
 	useEffect( () => {
 		// Require to start the flow from the first step
 		if ( ! selectedDesign ) {
-			goToStep?.( 'goals' );
+			if ( signupSelectedThemeSlug !== 'blank-canvas-3' ) {
+				goToStep?.( 'goals' );
+			}
+			// User has selected blank-canvas-3 theme from theme showcase
+			setSelectedDesign( blankCanvasDesign as Design );
```

Comment:
this has to be wrapped inside the `else` condition to ensure we won't save the blank canvas design after `goToStep` executed